### PR TITLE
server: do not repeatedly error with no such id

### PR DESCRIPTION
**Pull request by haircommander**
as it should not be a fatal error, as it indicates we've already cleaned up

Signed-off-by: Peter Hunt <pehunt@redhat.com>

#### Does this PR introduce a user-facing change?

```release-note
none
```

### CODE REVIEW

**Comment by haircommander** [comment]
similar to https://github.com/cri-o/cri-o/pull/4966

File: server/container_create.go [hunk 1]
```diff
@@ -495,6 +495,10 @@ func (s *Server) CreateContainer(ctx context.Context, req *types.CreateContainer
 		log.Infof(ctx, description)
 		err := s.CtrIDIndex().Delete(ctr.ID())
 		if err != nil {
+			// already deleted
+			if strings.Contains(err.Error(), "no such id") {
```
Comment (saschagrunert) [inline server/container_create.go:499]: Let's make `"no such id"` a `const` at least on our side.

**Review by saschagrunert — APPROVED** [review]
/retest

File: server/container_create.go (hunk 1 above)
Comment (haircommander) [reply server/container_create.go]: yeah good point

**Review by TomSweeneyRedHat — COMMENTED** [review]
LGTM
assuming you never want to see this error.  Based on the PR title, I thought you'd want to see the first instance, but not repeated ones.

File: server/container_create.go [hunk 2]
```diff
@@ -27,6 +27,9 @@ import (
 	"github.com/pkg/errors"
 )
 
+// sync with https://github.com/containers/storage/blob/7fe03f6c765f2adbc75a5691a1fb4f19e56e7071/pkg/truncindex/truncindex.go#L92
+const noSuchID = "no such id"
```
Comment (mrunalp) [inline server/container_create.go:31]: Why can't we have the storage code return this typed error instead - https://github.com/containers/storage/blob/7fe03f6c765f2adbc75a5691a1fb4f19e56e7071/pkg/truncindex/truncindex.go#L23?

Comment (haircommander) [reply server/container_create.go:31]: easier to backport this :) I planned on updating the error in main

**Comment by haircommander** [comment]
@saschagrunert PTAL
/override ci/prow/e2e-gcp

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Pull request merged**